Update change log common headings (#3103)
Azure/azure-sdk: docs/policies/releasenotes.md

We don't want to advertise every single bug fix as most of them do not impact the way customers think about developing the client.  The [change log](https://azure.github.io/azure-sdk/policies_releases.html#change-logs) provides an exhaustive list of changes.  We don't need to duplicate it.

> However, in the release notes we do want to list critical changes for customers. A critical change is one that the developer would either need to know or want to know. Use the following section headers (`Features Added`, `Breaking Changes`, and `Bugs Fixed`) for the defined critical changes:

Reviewer: 🎉

Reviewer 2: @weshaggard looks good to me.

> * *Features Added* - For new features to be called out in release notes

Author: ```suggestion
* *Features Added* - For new features to be called out in release notes. 
```

> * *Breaking Changes* - For changes to be called out in release notes including, changes that break existing functionality, soon-to-be removed features, now removed features

Author: ```suggestion
* *Breaking Changes* - For any changes that the will break the customer in some way, such as breaking existing functionality, soon-to-be removed features, or now removed features. 
```

> * *Features Added* - For new features to be called out in release notes. 

Reviewer 3: I assume this would need to be addressed separately, but it would be great if we can get this ghost text to populate in the Changelogs so we don't have to look up the guidelines every time we want to double check on something.

Author: I was trying to figure out how to provide some ghost text or links to changelog guidelines in the MD files but I don't know a good way to do it without breaking validation or without it accidently shipping in the changelog content. If you have some ideas on how to make that happen I'm open to them.

Reviewer 3: Don't we have validation right now that checks for empty sections? Could we also add a check for the ghost text, or are we worried that it might change and then no longer work?

Author: Exactly checking for exact text is dangerous because even a slight change would leave it behind. It would be a little safer to add our link in there and then validation can ensure that link doesn't exist in the content.

Reviewer 3: That makes sense - seeing as the sections will need to be updated anyway before release, I think having the link in there that would need to be deleted wouldn't be too much of an added burden.

Author: Lets consider that as a follow-up improvement once people start getting used to the template being in there maybe it won't be needed.

> * *Bugs Fixed* - For important bug fixes to be called out in release notes, including any security fixes

Author: ```suggestion
* *Bugs Fixed* - For important bug fixes to be called out in release notes, including any security fixes. Include things that the customer would likely notice or need to react to in some way. 
```

> For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the `Other Changes` heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this shouldn't be listed under the `Bugs Fixed` heading.  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

Reviewer 2: Yes, I like the category 'Other Changes' without the 'and Improvements' - which gets to be too long.

> For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the `Other Changes` heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this shouldn't be listed under the `Bugs Fixed` heading.  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

Reviewer 3: > However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

It would be good to clarify which bucket this would go in (Features Added)

> For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the `Other Changes` heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this shouldn't be listed under the `Bugs Fixed` heading.  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

Reviewer 3: > While you may list this under the `Other Changes` heading in the changelog, the customer does not need to do anything

I think this is probably the case for a lot of bugs that we fix.

>and it's fairly likely they have not bumped into this error

This seems to be the more salient point.

> For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the `Other Changes` heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this shouldn't be listed under the `Bugs Fixed` heading.  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

Reviewer 3: > For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.

I see this one as debatable. I think a really clear cut example is:
The name of the property displayed in the ArgumentOutOfRangeException in the MaxDeliveryCount property in SubscriptionProperties was updated to use the correct property name. (A community contribution, courtesy of oscarcabrero)

> For example, "fixed a bug in which the event processor would stop working if you received no events for 30 minutes" does not qualify.  While you may list this under the `Other Changes` heading in the changelog, the customer does not need to do anything, and it's fairly likely they have not bumped into this error, so this shouldn't be listed under the `Bugs Fixed` heading.  However, "Added a new overload to the constructor to support AzureAD credentials" would definitely be a good thing to include.

Author: ```suggestion
For example, "The name of the property displayed in the ArgumentOutOfRangeException in the MaxDeliveryCount property in SubscriptionProperties was updated to use the correct property name." does not qualify as an important bug fix so shouldn't be listed under `Bugs Fixed` but can be listed under `Other Changes`.  However, "Added a new overload to the constructor to support AzureAD credentials" would be a good thing to include under `Features Added`.
```
